### PR TITLE
IBX-154: Allowed to define if the alternative text for image field is required (2.5)

### PR DESCRIPTION
**Pull request by adamwojs**
| Question      | Answer
| ------------- | ---
| Tickets       | https://issues.ibexa.co/browse/IBX-154
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | ?
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

https://github.com/ezsystems/ezplatform-admin-ui/pull/1777 for v2.5

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

**Review by alongosz — APPROVED** [review]

File: src/bundle/Controller/AssetController.php [hunk 1]
```diff
@@ -84,7 +84,7 @@ public function uploadImageAction(Request $request): Response
                             'path' => $file->getRealPath(),
                             'fileSize' => $file->getSize(),
                             'fileName' => $file->getClientOriginalName(),
-                            'alternativeText' => null,
+                            'alternativeText' => $file->getClientOriginalName(),
```
Comment (Steveb-p) [inline src/bundle/Controller/AssetController.php:87]: Isn't this a possible XSS injection vector?

**Review by Steveb-p — APPROVED** [review]

**Comment by piotrbisaga** [comment]
At first glance I saw that new checkbox "alternative text is required" is not written with bold when comparing it to "required" and "translatable"
Screenshot 2021-06-22 at 09 08 51

**Comment by adamwojs** [comment]
@piotrbisaga https://github.com/ezsystems/ezplatform-admin-ui/pull/1782/commits/fbe6173d533fd0548b0aa71466bf0283835c2619 should fix this inconsistency

**Comment by piotrbisaga** [comment]
Please take a look at the below described output after performing 2/4 possible options in Jira

1/
On the left nr2: image field required and alternative text required
On the right nr4: image filed not required and alternative text required

Comparison only to highlight difference in exclamation mark colour and difference between validating errors.

2/ 
After applying step nr4: uploaded image disappears right after clicking "Upload file"

Screenshot 2021-06-22 at 13 10 13

**Review by piotrbisaga — APPROVED** [review]
QA approved, checked on ez Platform Enterprise 2.5 with .diff

**Pull request merged**